extension.go

```diff
@@ -8,6 +8,51 @@ import (
 	"github.com/golang/protobuf/proto"
 )
 
+// An Extension is a custom option annotation that can be applied to an Entity to provide additional
+// semantic details and metadata about the Entity.
+type Extension interface {
+	Field
+
+	// ParentEntity returns the ParentEntity where the Extension is defined
+	ParentEntity() ParentEntity
```

Comment:
Let's go with something like `DefinedIn()` to make it clear from the function name what the returned value represents, without having to look up the definition.